pkg/repository/sequence/mongo.go

```diff
@@ -0,0 +1,176 @@
+package sequence
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var _ Repository = (*MongoRepository)(nil)
+
+const (
+	CollectionSequences = "sequences"
+)
+
+type ExecutionType string
+
+const (
+	ExecutionTypeTest         ExecutionType = "t"
+	ExecutionTypeTestSuite    ExecutionType = "ts"
+	ExecutionTypeTestWorkflow ExecutionType = "tw"
+)
+
+func NewMongoRepository(db *mongo.Database, opts ...Opt) *MongoRepository {
+	r := &MongoRepository{
+		Coll: db.Collection(CollectionSequences),
+	}
+
+	for _, opt := range opts {
+		opt(r)
+	}
+
+	return r
+}
+
+type Opt func(*MongoRepository)
+
+func WithMongoRepositoryCollection(collection *mongo.Collection) Opt {
+	return func(r *MongoRepository) {
+		r.Coll = collection
+	}
+}
+
+type MongoRepository struct {
+	Coll *mongo.Collection
+}
+
+type oldExecutionNumber struct {
+	Name        string `json:"name"`
+	Number      int    `json:"number"`
+	IsTestSuite bool   `json:"isTestSuite"`
+}
+
+type executionNumber struct {
+	Id            string        `bson:"_id"`
+	Number        int           `bson:"number"`
+	ExecutionType ExecutionType `bson:"executionType"`
+}
+
+// GetNextExecutionNumber gets next execution number by name and type
+func (r *MongoRepository) GetNextExecutionNumber(ctx context.Context, name string, executionType ExecutionType) (number int32, err error) {
```

Comment:
This have a downside, that the Test / Test Workflow etc won't necessarily be created after `GetNextExecutionNumber`, so consecutive executions won't have consecutive sequence numbers.

From a technical perspective it's rather a better situation than before, but from the user perspective - the gaps between executions will lead to questions.

Rather creation + sequence increment should be atomic.

Reply:
consecutive is a bit weird in parallel world. if we run 100 executions in parallel, the number doesn't mean this much. still we can  create a transaction on inserting executtion and getting the number to avoid gaps

Reply:
transactions in mongo are supported only for  replicaset, we can't guarantee